src/lib/content.js

```diff
@@ -0,0 +1,171 @@
+import marked from 'marked';
+
+// Find YAML FrontMatter preceeding a markdown document
+const FRONT_MATTER_REG = /^\s*---\n\s*([\s\S]*?)\s*\n---\n/i;
+
+const MARKDOWN_TITLE = /(?:^|\n\n)\s*(#{1,6})\s+(.+)\n+/g;
+
+// Store URL-keyed cached Promise return values from getContent in production.
+const CACHE = {};
+
+/**
+ * Fetch and parse a markdown document with optional JSON FrontMatter.
+ * @returns {{ content: string, meta: {toc:{text:string, id:string, level:number}[], title: string} }}
+ */
+export function getContent([lang, name]) {
+	let path = `/content/${lang}`,
+		url = `${path}/${name.replace(/^\//, '')}`,
+		ext = (url.match(/\.([a-z]+)$/i) || [])[1];
+	if (!ext) url += '.md';
+
+	// In prod, never re-fetch the content (url is just a convenient compound cache key)
+	if (process.env.NODE_ENV === 'production' && url in CACHE) {
+		return CACHE[url];
+	}
+
+	let fallback = false;
+	const res = fetch(url)
+		.then(r => {
+			// fall back to english
+			if (!r.ok && lang != 'en') {
+				fallback = true;
+				return fetch(url.replace(/content\/[^/]+\//, 'content/en/'));
+			}
+			return r;
+		})
+		.then(r => {
+			if (r.ok) return r;
+			ext = 'md';
+			return fetch(`${path}/${r.status}.md`);
+		})
+		.then(r => r.text())
+		.then(parseContent)
+		.then(applyEmojiToContent)
+		.then(data => {
+			data.fallback = fallback;
+			return data;
+		});
```

Comment:
Should we add a `catch` handler to clear the cache when the request failed? A hard reload would do the same thing so I'm not sure if this is necessary.